File: 02_02_2024_Sequential_Digits.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        String s = "123456789";
        List<Integer> ans = new ArrayList<>();

        for (int i = 0; i < s.length(); i++) {
            StringBuilder st = new StringBuilder();
            for (int j = i; j < s.length(); j++) {
                st.append(s.charAt(j));
                int num = Integer.parseInt(st.toString());
                if (num >= low && num <= high) {
                    ans.add(num);
                }
            }
        }

        Collections.sort(ans);
        return ans;
    }
}

File: 02_02_2024_Sequential_Digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        string s="123456789";
        vector<int> ans;
        for(int i=0;i<s.size();i++){
            string st="";
            for(int j=i;j<s.size();j++){
                    st+=s[j];
                    int num=stoi(st);
                    if(num>=low&&num<=high){
                        ans.push_back(num);
                    }
            }
        }
        sort(ans.begin(),ans.end());
        return ans;
    }
};
